src/NewsService.Data/NewsRepository.cs

```diff
@@ -62,38 +80,13 @@ public List<DbNews> Find(FindNewsFilter findNewsFilter, out int totalCount)
         return null;
       }
 
-      if (findNewsFilter.SkipCount < 0)
-      {
-        totalCount = 0;
-        return null;
-      }
-
-      if (findNewsFilter.TakeCount < 1)
-      {
-        totalCount = 0;
-        return null;
-      }
-
       IQueryable<DbNews> dbNewsList = _provider.News.AsQueryable();
 
-      if (findNewsFilter.AuthorId.HasValue)
-      {
-        dbNewsList = dbNewsList.Where(x => x.AuthorId == findNewsFilter.AuthorId);
-      }
-
-      if (findNewsFilter.DepartmentId.HasValue)
-      {
-        dbNewsList = dbNewsList.Where(x => x.DepartmentId == findNewsFilter.DepartmentId);
-      }
-
-      if (!findNewsFilter.IncludeDeactivated)
-      {
-        dbNewsList = dbNewsList.Where(x => x.IsActive);
-      }
+      CreateFindPredicates(findNewsFilter, dbNewsList).FirstOrDefault();
```

Comment:
i think it strange that we change dbNewsList via parameter. I think we need to return it. or then i dont understand why we need to return value from this func. To make FristOrDefault? for what if we dont save a result?

Reply:
really smth strange here

Reply:
done

Reply:
done